docs/src/man/split_apply_combine.md

```diff
@@ -122,11 +122,13 @@ It is allowed to mix single values and vectors if multiple transformations
 are requested. In this case single value will be repeated to match the length
 of columns specified by returned vectors.
 
-A separate task is spawned for each specified transformation, allowing for
-parallel operation when several transformations are requested and Julia was
-started with more than one thread. Passed transformation functions should
-therefore not modify global variables (i.e. they should be pure), or use
-locks to control parallel accesses.
+A separate task is spawned for each specified transformation; each transformation
+then spawns as many tasks as Julia threads, and splits processing of groups across
+them (however, currently transformations with optimized implementations like `sum`
```

Comment:
so this means that e.g. when we have two single-row transformations we spin up `2*Threads.nthreads()` tasks. I guess (hope :smile:) it is OK, but just want to make sure.

Reply:
Yes. We could keep track of how many transformations have been passed (and whether some of them are fast aggregations), but it sounds OK to spawn more tasks than needed. That could be faster as not all transformations are likely to take the same time. AFAIK tasks are supposed to be lightweight and we could start one task per group without killing Julia. :-)

Reply:
This is not my experience from the past (but maybe this has changed). I remember that spawning many tasks was negligible (benchmarks should show :smile: ).